ccl/streamingccl/streamproducer: TestStreamAddSSTable failed

ccl/streamingccl/streamproducer.TestStreamAddSSTable [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9881217?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_StressBazel/9881217?buildTab=artifacts#/) on master @ [7d02e061759599a2b054ee06c71524443f77d20a](https://github.com/cockroachdb/cockroach/commits/7d02e061759599a2b054ee06c71524443f77d20a):


```
=== RUN   TestStreamAddSSTable
    test_log_scope.go:161: test logs captured to: /artifacts/tmp/_tmp/9ce24496530642208197dda818d65c9e/logTestStreamAddSSTable1120974889
    test_log_scope.go:79: use -show-logs to present logs inline
    replication_stream_test.go:565: -- test log scope end --
    replication_stream_test.go:565: Leaked goroutine: goroutine 5812317 [select]:
        github.com/jackc/pgconn/internal/ctxwatch.(*ContextWatcher).Watch.func1()
        	github.com/jackc/pgconn/internal/ctxwatch/external/com_github_jackc_pgconn/internal/ctxwatch/context_watcher.go:47 +0x6c
        created by github.com/jackc/pgconn/internal/ctxwatch.(*ContextWatcher).Watch
        	github.com/jackc/pgconn/internal/ctxwatch/external/com_github_jackc_pgconn/internal/ctxwatch/context_watcher.go:46 +0x12c
        Leaked goroutine: goroutine 5829131 [IO wait]:
        internal/poll.runtime_pollWait(0x7fef78b901f0, 0x72)
        	GOROOT/src/runtime/netpoll.go:305 +0x89
        internal/poll.(*pollDesc).wait(0xc01651ff80?, 0xc0106c2000?, 0x0)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:84 +0x32
        internal/poll.(*pollDesc).waitRead(...)
        	GOROOT/src/internal/poll/fd_poll_runtime.go:89
        internal/poll.(*FD).Read(0xc01651ff80, {0xc0106c2000, 0x10, 0x10})
        	GOROOT/src/internal/poll/fd_unix.go:167 +0x25a
        net.(*netFD).Read(0xc01651ff80, {0xc0106c2000?, 0xc007a6fe48?, 0x4a7a92?})
        	GOROOT/src/net/fd_posix.go:55 +0x29
        net.(*conn).Read(0xc00f5e8008, {0xc0106c2000?, 0xc005227680?, 0xc005227680?})
        	GOROOT/src/net/net.go:183 +0x45
        github.com/jackc/pgconn.(*PgConn).CancelRequest(0xc0050a6840, {0x75387e8?, 0xc005227680})
        	github.com/jackc/pgconn/external/com_github_jackc_pgconn/pgconn.go:934 +0x3db
        github.com/jackc/pgconn.(*PgConn).asyncClose.func1()
        	github.com/jackc/pgconn/external/com_github_jackc_pgconn/pgconn.go:667 +0x10f
        created by github.com/jackc/pgconn.(*PgConn).asyncClose
        	github.com/jackc/pgconn/external/com_github_jackc_pgconn/pgconn.go:658 +0x65
--- FAIL: TestStreamAddSSTable (25.63s)
```
<p>Parameters: <code>TAGS=bazel,gss,deadlock</code>
</p>
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/disaster-recovery
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestStreamAddSSTable.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-27582

--- a/pkg/util/leaktest/leaktest.go
+++ b/pkg/util/leaktest/leaktest.go
@@ -58,6 +58,10 @@ func interestingGoroutines() map[int64]string {
 			strings.Contains(stack, "sentry-go.(*HTTPTransport).worker") ||
 			// Ignore the opensensus worker, which is created by the event exporter.
 			strings.Contains(stack, "go.opencensus.io/stats/view.(*worker).start") ||
+			// Ignore pgconn which creates a goroutine to do an async cleanup.
+			strings.Contains(stack, "github.com/jackc/pgconn.(*PgConn).asyncClose.func1") ||
+			// Ignore pgconn which creates a goroutine to watch context cancellation.
+			strings.Contains(stack, "github.com/jackc/pgconn/internal/ctxwatch.(*ContextWatcher).Watch.func1") ||
 			// Seems to be gccgo specific.
 			(runtime.Compiler == "gccgo" && strings.Contains(stack, "testing.T.Parallel")) ||
 			// Ignore intentionally long-running logging goroutines that live for the

